task20/js/main.js

```diff
@@ -0,0 +1,88 @@
+import '../css/style.css'
+import { createElementWithClassName } from './utils/createElement'
+import { renderLoading } from './modules/loading'
+import { removeLoading } from './modules/loading'
+import { renderErrorMessage } from './modules/error'
+
+const endpoint = "https://mocki.io/v1/7afded4f-46ed-4535-b770-997abc352aea";
+
+const parent = document.getElementById("js-parent");
+
+const getFetchData = async () => {
+    const response = await fetch(endpoint);
+    if (!response.ok){
+        console.error(`$(response:status):$(response:statusText)`);
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    }
+    const json = await response.json();
+    return json.data;
+};
+
+const fetchTableData = () => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(getFetchData(endpoint));
+        }, 3000);
+    });
+};
+
+const initialize = async () => {
+    renderLoading(parent);
+    let userContentsData
+    try {
+        userContentsData = await fetchTableData();
+    } catch(error) {
+        renderErrorMessage(parent, "問題が発生し表示することができません。");
+    } finally {
+        removeLoading(parent);
+    }
+    renderTableContents(userContents, userContentsData);
+};
```

Comment:
IMO
If the json data is empty, it would be better if a message could be output.

- empty json data
https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236

Reply:
Is this the correct way to write this? I have never written errors for lack of data before, so this was a good learning experience. Thank you very much.
[f48fd84](https://github.com/seeeen044/JavaScript_handson/pull/21/commits/f48fd849a0571f9a1d63f499d1c14c6606f2ea89)

Reply:
I wrote a new comment below. Sorry!